src/format/OpData01.cpp

```diff
@@ -0,0 +1,143 @@
+#include "OpData01.h"
+
+#include "crypto/CryptoHash.h"
+#include "crypto/SymmetricCipher.h"
+
+#include <QMessageLogger>
+#include <QtGui/QtGui> // for qWarning() et al
+
+#define RETURN_OPDATA01_ERROR_INSTANCE(msg)                                                                            \
```

Comment:
Do not use a macro for returns. The generally accepted solution is to have a QString passed in as reference to the function with the return type of bool indicating success/failure. Also this creates a mess of dangling pointers if they aren't handled correctly.